CalibFormats/SiPixelObjects/interface/PixelCalibBase.h

```diff
@@ -28,7 +28,7 @@ namespace pos{
     PixelCalibBase();
     virtual ~PixelCalibBase();
     virtual std::string mode() const {return mode_;}
-    virtual void writeXMLHeader(  pos::PixelConfigKey &key, 
+    virtual void writeXMLHeader(  pos::PixelConfigKey key, 
```

Comment:
No classes which inherit from this base class ever override this function. Instead, most use multiple inheritance to inherit from another base class which has a member function with the same name but passes the first argument by value. I couldn't find anywhere that this base class is used directly (i.e. via pointer or reference). Therefore I think it is safe to completely remove this function from this class.